csharp/src/Ice/Connection.cs

```diff
@@ -988,7 +971,8 @@ public override void Message(ref ThreadPoolCurrent current)
 
                                 if (_endpoint.Datagram() && size > _readBufferOffset)
                                 {
-                                    throw new DatagramLimitException(); // The message was truncated.
+                                    throw new DatagramLimitException(
```

Comment:
Instead of throwing, we could just return here after logging the warning and reseting the read buffer like we do in the catch block.

Reply:
Sounds good, fixing as suggested.